### PR TITLE
Use Umpire's `QuickPool` and change Umpire version requirement

### LINKED ISSUE #420: Compilation issues with Umpire@6.0.0
As pointed out by @albestro :

DynamicPool & DynamicPoolMap seem to be deprecated in 6.0.0:
 https://github.com/LLNL/Umpire/pull/635
 
Changelog : https://github.com/LLNL/Umpire/blob/develop/CHANGELOG.md#changed-1

```
     150    In file included from /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:4:
  >> 151    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/strategy/DynamicPool.hpp:16:1: error: expected unqual
            ified-id before 'using'
     152     using DynamicPool = DynamicPoolMap;
     153     ^~~~~
     154    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp: In function 'void dlaf::memory::internal::initializeUmpireHostAllocato
            r(std::size_t)':
  >> 155    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:46: error: 'DynamicPool' is not a member of 'umpire::strategy'
     156                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     157                                                  ^~~~~~~~~~~
     158    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:46: note: suggested alternative: 'DynamicPoolMap'
     159                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     160                                                  ^~~~~~~~~~~
     161                                                  DynamicPoolMap
  >> 162    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:46: error: 'DynamicPool' is not a member of 'umpire::strategy'
     163    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:46: note: suggested alternative: 'DynamicPoolMap'
     164                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     165                                                  ^~~~~~~~~~~
     166                                                  DynamicPoolMap
  >> 167    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:103: error: no matching function for call to 'umpire::ResourceManage
            r::makeAllocator<<expression error> >(const char [12], umpire::Allocator&, std::size_t&)'
     168                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     169                                                                                                           ^
     170    In file included from /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:3:
     171    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:117:13: note: candidate: 'templat
            e<class Strategy, bool introspection, class ... Args> umpire::Allocator umpire::ResourceManager::makeAllocator(const string&, Args&& ...)'
     172       Allocator makeAllocator(const std::string& name, Args&&... args);
     173                 ^~~~~~~~~~~~~
     174    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:117:13: note:   template argument
             deduction/substitution failed:
  >> 175    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:103: error: template argument 1 is invalid
     176                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     177                                                                                                           ^
     178    In file included from /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:3:
     179    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:120:13: note: candidate: 'templat
            e<class Strategy, class ... Args> umpire::Allocator umpire::ResourceManager::makeAllocator(const string&, umpire::Tracking, Args&& ...)'
     180       Allocator makeAllocator(const std::string& name, Tracking tracked, Args&&... args);
     181                 ^~~~~~~~~~~~~
     182    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:120:13: note:   template argument
             deduction/substitution failed:
  >> 183    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:103: error: template argument 1 is invalid
     184                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     185                                                                                                           ^
     186    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp: In function 'void dlaf::memory::internal::initializeUmpireDeviceAlloca
            tor(std::size_t)':
  >> 187    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:64:46: error: 'DynamicPool' is not a member of 'umpire::strategy'
     188                 .makeAllocator<umpire::strategy::DynamicPool>("DEVICE_pool", device_allocator,
     189                                                  ^~~~~~~~~~~
     190    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:64:46: note: suggested alternative: 'DynamicPoolMap'
     191                 .makeAllocator<umpire::strategy::DynamicPool>("DEVICE_pool", device_allocator,
     192                                                  ^~~~~~~~~~~
     193                                                  DynamicPoolMap
  >> 194    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:64:46: error: 'DynamicPool' is not a member of 'umpire::strategy'
     195    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:64:46: note: suggested alternative: 'DynamicPoolMap'
     196                 .makeAllocator<umpire::strategy::DynamicPool>("DEVICE_pool", device_allocator,
     197                                                  ^~~~~~~~~~~
     198                                                  DynamicPoolMap
  >> 199    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:65:72: error: no matching function for call to 'umpire::ResourceManager
            ::makeAllocator<<expression error> >(const char [12], umpire::Allocator&, std::size_t&)'
     200                                                               initial_bytes);
     201                                                                            ^
     202    In file included from /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:3:
     203    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:117:13: note: candidate: 'templat
            e<class Strategy, bool introspection, class ... Args> umpire::Allocator umpire::ResourceManager::makeAllocator(const string&, Args&& ...)'
     204       Allocator makeAllocator(const std::string& name, Args&&... args);
     205                 ^~~~~~~~~~~~~
     206    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:117:13: note:   template argument
             deduction/substitution failed:
  >> 207    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:65:72: error: template argument 1 is invalid
     208                                                               initial_bytes);
     209                                                                            ^
     210    In file included from /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:3:
     211    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:120:13: note: candidate: 'templat
            e<class Strategy, class ... Args> umpire::Allocator umpire::ResourceManager::makeAllocator(const string&, umpire::Tracking, Args&& ...)'
     212       Allocator makeAllocator(const std::string& name, Tracking tracked, Args&&... args);
     213                 ^~~~~~~~~~~~~
     214    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:120:13: note:   template argument
             deduction/substitution failed:
  >> 215    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:65:72: error: template argument 1 is invalid
     216                                                               initial_bytes);
     217                                                                            ^
  >> 218    make[2]: *** [src/CMakeFiles/dlaf.core.dir/build.make:233: src/CMakeFiles/dlaf.core.dir/memory/memory_chunk.cpp.o] Error 1
     219    make[2]: *** Waiting for unfinished jobs....
     220    make[2]: Leaving directory '/tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-build-tuyamvo'
     221    [ 75%] Built target dlaf.auxiliary
     222    make[2]: Leaving directory '/tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-build-tuyamvo'
  >> 223    make[1]: *** [CMakeFiles/Makefile2:1021: src/CMakeFiles/dlaf.core.dir/all] Error 2
     224    make[1]: *** Waiting for unfinished jobs....
     225    make[2]: Leaving directory '/tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-build-tuyamvo'
     226    [ 75%] Built target dlaf.solver
     227    make[2]: Leaving directory '/tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-build-tuyamvo'
     228    [ 75%] Built target dlaf.factorization
     229    make[2]: Leaving directory '/tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-build-tuyamvo'
```

### PR DESCRIPTION
**Pull request by aurianer**
Fixes #420.

### CODE REVIEW

File: spack/packages/dla-future/package.py [hunk 1]
```diff
@@ -44,9 +44,6 @@ class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("umpire+cuda~shared", when="+cuda")
     depends_on("umpire+rocm~shared", when="+rocm")
```
Comment (msimberg) [inline spack/packages/dla-future/package.py:46]: Please add a minimum version requirement to umpire above.

Comment (msimberg) [reply spack/packages/dla-future/package.py:46]: > Should I check which one is the first to have it?

Yes please (also check that the header that includes it is the same).

**Comment by msimberg** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Review by rasolca — APPROVED** [review]

**Pull request merged**